src/doc/src/reference/build-scripts.md

```diff
@@ -250,6 +255,51 @@ identifier, the value should be a string.
 [conditional compilation]: ../../reference/conditional-compilation.md
 [option-cfg]: ../../rustc/command-line-arguments.md#option-cfg
 
+### `cargo::rustc-check-cfg=CHECK_CFG` {#rustc-check-cfg}
+
+This instruction is related to checking the conditional configurations of the
+crate at compile time. That is, checking that every _reachable_ `#[cfg]` matches
+the expected config names and values.
+
+The `rustc-check-cfg` instruction adds to the list of expected config names
+and values in addition to Cargo features and [well known cfgs][well-known-cfgs].
+
+It is backed by the [`--check-cfg` flag][option-check-cfg] of the compiler.
```

Comment:
Instead, could we say that the syntax is what `--check-cfg` accepts, like other instructions do?

> The rustc-link-arg instruction tells Cargo to pass the [-C link-arg=FLAG option](https://doc.rust-lang.org/nightly/rustc/codegen-options/index.html#link-arg) to the compiler,

Reply:
See this [thread](https://github.com/rust-lang/cargo/pull/13571#discussion_r1569517868) with `@epage` about why it differ, but the basic idea was to put the Why? first and then the How?.